src/utils/aggregate.js

```diff
@@ -0,0 +1,43 @@
+import d3 from './d3.js';
+
+/**
+ * Get aggregated values using specified function, such as a mean of quantitative values.
+ * @param {object|array} rowInfo Object or array of objects containing information of rows.
+ * @param {string} field The name of data field.
+ * @param {string} type The type of data field.
+ * @param {string} fn A function to aggregate values.
+ */
+export function getAggregatedValue(rowInfo, field, type, fn = "sum") {
+    const isAggregated = Array.isArray(rowInfo);
+    if(!isAggregated) {
+        return rowInfo[field];
+    }
+
+    // Aggregated values
+    if(type === "nominal" || type === "link") {
+        if(fn === "max") {
+            const counts = {};
+            rowInfo.forEach(d => {
+                counts[d[field]] = 1 + (counts[d[field]] || 0);
+            });
+            const topCategory = Object.keys(counts).reduce((a, b) => counts[a] > counts[b] ? a : b);
+            return topCategory;
+        } else {
+            const categories = Array.from(new Set(rowInfo.map(d => d[field])));
+            return categories.join(", ");
+        } 
+    } else if(type === "quantitative") {
```

Comment:
Rather than `if else if else` statements, can the functions be defined in an object like

```js
const aggFunctionMapping = {
    quantitative: {
        max: d3.max,
        mean: d3.mean,
        sum: d3.sum
    },
    tree: {
        identity: (d) => d,
    }
};
```

and accessed like `aggFunctionMapping[type][functionName]`

Reply:
I like this approach better! This looks more compact and readable.